Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -2058,7 +2058,10 @@ void WebPageProxy::addPlatformLoadParameters(WebProcessProxy&, LoadParameters&)
 
 WebProcessProxy& WebPageProxy::ensureRunningProcess()
 {
-    if (!hasRunningProcess())
+    if (m_isClosed)
+        WEBPAGEPROXY_RELEASE_LOG_ERROR(Loading, "ensureRunningProcess: Page is closed");
+
+    if (m_legacyMainFrameProcess->wasTerminated() && !m_isClosed)
```

Comment:
Why is this no longer relying on `hasRunningProcess()`?

Also, technically, this is a bug at the call site. Can we fix the call site instead of add a `!m_isClosed` assertion here? It makes no sense to ask for a process after a WKWebView has been closed.

Reply:
I don't like that it hides bugs in callers. As evidence by this PR where you don't even fix the call sites, people will stop caring and do wrong things because it kind of works.